### PR TITLE
Remove PG_REGISTRY_TAIL_SECTION from configuration storage

### PR DESCRIPTION
**Pull request by hydra**
@ledvinap This is a solution to your comment regarding the registry tail section.

https://github.com/cleanflight/cleanflight/pull/1976#discussion_r57434804

Please review and comment.

I don't like it from a unit test perspective.  is there a better way to do it?

### CODE REVIEW

File: src/main/target/stm32_flash.ld [hunk 1]
```diff
@@ -82,12 +84,13 @@ SECTIONS
...
     PROVIDE_HIDDEN (__fini_array_end = .);
   } >FLASH
+  
   .pg_registry :
   {
     PROVIDE_HIDDEN (__pg_registry = .);
     KEEP (*(.pg_registry))
     KEEP (*(SORT(.pg_registry.*)))
-    KEEP (*(.pg_registry_tail))
+    PROVIDE_HIDDEN (__pg_registry_end = .);
```
Comment (hydra) [inline src/main/target/stm32_flash.ld]: @ledvinap is this how you meant to adjust the linker script?

File: src/test/unit/config_eeprom_unittest.cc [hunk 2]
```diff
@@ -99,17 +100,17 @@ const pgRegistry_t __pg_registry[] =
...
-        .ptr = 0,
-        .size = 0,
-        .pgn = 0,
-        .format = 0,
-        .flags = 0
-    },
+    }
 };
 
+const pgRegistry_t __pg_registry_end[] = {}; // FIXME enforce the location of this somehow.
```
Comment (hydra) [inline src/test/unit/config_eeprom_unittest.cc]: @ledvinap this is specifically what i do not like about this solution currently.

**Comment by ledvinap** [comment]
https://github.com/cleanflight/cleanflight/pull/1981 (first commit) does implement registry start/end, including unittest (inserting section into default ld script, works both on gcc and clang)

**Comment by hydra** [comment]
superseded by #1981

**Pull request closed without merging**